packages/contracts/src/token/ERC20/governance/GovernanceERC20.sol

```diff
@@ -111,8 +111,9 @@ contract GovernanceERC20 is
     /// @inheritdoc ERC20VotesUpgradeable
     function _afterTokenTransfer(address from, address to, uint256 amount) internal override {
         super._afterTokenTransfer(from, to, amount);
-        // reduce _delegate calls only when minting
-        if (from == address(0) && to != address(0) && delegates(to) == address(0)) {
+
+        // Automatically turn on delegation on mint/transfer but only for the first time.
+        if (to != address(0) && numCheckpoints(to) == 0) {
```

Comment:
I am wondering if this will ever be true for a transfer send with `from != to` and `amount > 0` (the case of an acutal transfer.

The functions calls `super._afterTokenTransfer(from, to, amount);` which redirects it to `_afterTokenTransfer` located in `ERC20VotesUpgradeable`

```solidity
function _afterTokenTransfer(
    address from,
    address to,
    uint256 amount
) internal virtual override {
    super._afterTokenTransfer(from, to, amount);
    _moveVotingPower(delegates(from), delegates(to), amount);
}
```
Next, this triggers `_moveVotingPower(delegates(from), delegates(to), amount);` in `ERC20VotesUpgradeable`
```solidity
function _moveVotingPower(
    address src,
    address dst,
    uint256 amount
) private {
    if (src != dst && amount > 0) {
        if (src != address(0)) {
            (uint256 oldWeight, uint256 newWeight) = _writeCheckpoint(_checkpoints[src], _subtract, amount);
            emit DelegateVotesChanged(src, oldWeight, newWeight);
        }
        if (dst != address(0)) {
            (uint256 oldWeight, uint256 newWeight) = _writeCheckpoint(_checkpoints[dst], _add, amount);
            emit DelegateVotesChanged(dst, oldWeight, newWeight);
        }
    }
}
```
Here, because `src != dst` and `src != address(0)` both are true, we end up in the first `if` branch where `_writeCheckpoint(_checkpoints[dst], _add, amount);` happens so that a checkpoint is actually written for `to`.

Accordingly and if I didn't make a mistake, the second condition `if (to != address(0) && numCheckpoints(to) == 0)` in the implementation above should never be true.

Reply:
Removing the `if (to != address(0) && numCheckpoints(to) == 0)` leads a couple of tests to fail which means that I am wrong (which is good :) ). However, I am still wondering why it works.

Reply:
The explanation:

The first `_moveVotingPower(delegates(from), delegates(to), amount);` doesn't do anything for someone who hasn't delegated tokens yet (`delegates(to) == address(0)`)